File: C_Prog/binary_search.c
#include<stdio.h>
int main()
{
    int i,l,j,n,mid,key;
    printf("enter the length of array:");
    scanf("%d",&l);
    int a[l];
    printf("enter the element:");
    scanf("%d",&n);
    for (i=0;i<l;i++)
    {
        a[i]=i+1;
    }
    mid=l/2;
    for (i=0;i<l;i++)
    {
        key=a[i];
        j=i-1;
        while (a[j]!=a[i] && j<l)
        {
            a[i]=a[j];

        }
        a[i]=key;
    }
    for (i=0;i<l;i++)
    {
        printf("%d \n",a[i]);
    }
    return 0;
}

File: Py_prog/binary_search.py
n=int(input("enter the legth of the list:"))
l=[int(input()) for x in range(n)]

print('the elements in the list are:')
for i in l:
    print(i,end=' ')

e=int(input("enter the element to be searched:"))
mid=n//2
c=0
temp=n
#while(l[mid]!=e):
while(temp!=1):
    if l[mid]>e:
        mid=mid//2
    elif l[mid]==e:
        print("the element fount at ",mid)
        break
    else:
        mid+=mid//2
    temp//=2
    c+=1

if temp==1 and l[mid]!=e:
    print('the element not found')
#print("the element has found at ",mid)

print("no of iterations are:",c)